File: 532-k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.cpp
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        vector<int> v;
        unordered_map<int, int> mp, unique;
        int counter = 0;

        if(k == 0) {
            for(int it: nums)
                unique[it]++;
            
            for(auto it: unique)
                if(it.second > 1)
                    counter++;
            
            return counter;
        }
        for(int it: nums)
            unique[it] = 1;
        
        for(int i = 0; i < nums.size(); i++)
            if(unique[nums[i]] == 1)
                unique[nums[i]] = 0;
            else    
                nums[i] = -1e9;

        for(int it: nums)
            if(it != -1e9)
                v.push_back(it);

        for(int it: v) {
            int  above = it + k, below = it - k;

            if(mp.find(above) != mp.end())
                counter++;
            if(mp.find(below) != mp.end())
                counter++;
            
            mp[it]++;
        }

        return counter;
    }
};

File: 532-k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        l1=list(set(nums))
        count=0
        d=defaultdict(int)
        for i in l1:
            t1,t2=i+k,i-k
            if t1 in d:
                count+=d[t1]
            if t2 in d:
                count+=d[t2]  
            d[i]+=1
        if(k==0):
            t=defaultdict(int)
            for i in nums:
                t[i]+=1
            count=0
            for i in t:
                if(t[i]>=2):
                    count+=1    
        return count          
        